engine/engine.go

```diff
@@ -319,6 +327,25 @@ func (e *Engine) ReverseProxy(w http.ResponseWriter, r *http.Request, target *ur
 		if contentTypeOverwrite != "" {
 			proxyRes.Header.Set(HeaderContentType, contentTypeOverwrite)
 		}
+		if contentType := proxyRes.Header.Get(HeaderContentType); contentType == MediaTypeJSON && validateResponse {
+			var reader io.Reader
+			var err error
+			if proxyRes.Header.Get(HeaderContentEncoding) == FormatGzip {
+				reader, err = gzip.NewReader(proxyRes.Body)
+				if err != nil {
+					log.Printf("%v", err.Error())
+					return err
+				}
+			} else {
+				reader = proxyRes.Body
+			}
+			res, err := io.ReadAll(reader)
```

Comment:
Ik vraag mij af of de body nog moet worden geclosed. We hebben daar een linter voor (https://github.com/timakin/bodyclose?tab=readme-ov-file#analyzer). Maar je zit hier natuurlijk in een proxy dus dan is het wellicht anders. Mss kan het even proberen?

Reply:
Linting geeft sowieso geen errors, maar een `defer reader.Close()` kan sowieso geen kwaad, voeg het even toe.